src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -190,20 +188,41 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
     private ResponseList<T> getMany(Request req, Response res) {
         int limit = HttpUtils.getQueryParamFromRequest(req, LIMIT_PARAM, 0, DEFAULT_LIMIT, 100);
         int offset = HttpUtils.getQueryParamFromRequest(req, OFFSET_PARAM, 0, DEFAULT_OFFSET);
-        Auth0UserProfile requestingUser = getUserFromRequest(req);
-        if (isUserAdmin(requestingUser)) {
+        String userId = HttpUtils.getQueryParamFromRequest(req, USER_ID_PARAM, true);
+        // Filter the response based on the user id, if provided.
+        // If the user id is not provided filter response based on requesting user.
+        RequestingUser requestingUser = Auth0Connection.getUserFromRequest(req);
+        if (userId != null) {
+            OtpUser otpUser = Persistence.otpUsers.getById(userId);
+            if (otpUser != null && otpUser.canBeManagedBy(requestingUser)) {
+                return persistence.getResponseList(Filters.eq(USER_ID_PARAM, userId), offset, limit);
+            } else {
+                res.status(HttpStatus.FORBIDDEN_403);
+                return null;
+            }
+        }
+        if (requestingUser.isAdmin()) {
             // If the user is admin, the context is presumed to be the admin dashboard, so we deliver all entities for
             // management or review without restriction.
             return persistence.getResponseList(offset, limit);
         } else if (persistence.clazz == OtpUser.class) {
             // If the required entity is of type 'OtpUser' the assumption is that a call is being made via the
-            // OtpUserController. Therefore, the request should be limited to return just the entity matching the
-            // requesting user.
-            return persistence.getResponseList(Filters.eq("_id", requestingUser.otpUser.id), offset, limit);
+            // OtpUserController. If the request is being made by an Api user the response will be limited to the Otp users
+            // created by this Api user. If not, the assumption is that an Otp user is making the request and the response
+            // will be limited to just the entity matching this Otp user.
+            return (requestingUser.apiUser != null)
+                ? persistence.getResponseList(Filters.eq("applicationId", requestingUser.apiUser.id), offset, limit)
+                : persistence.getResponseList(Filters.eq("_id", requestingUser.otpUser.id), offset, limit);
```

Comment:
Minor syntax thing, but I think it's more readable to use the tertiary to define the filter and then only call `persistence.getResponseList` one time in this conditional block.

Reply:
Updated.